jaxrs-clients/src/test/java/com/palantir/remoting3/jaxrs/feignimpl/UserAgentTest.java

```diff
@@ -58,9 +60,25 @@ public void testUserAgent_default() throws InterruptedException {
     }
 
     @Test
-    public void testUserAgent_invalidUserAgentThrows() throws InterruptedException {
-        expectedException.expect(IllegalArgumentException.class);
-        expectedException.expectMessage(is("User Agent must match pattern '[A-Za-z0-9()\\-#;/.,_\\s]+': !@"));
-        JaxRsClient.create(TestService.class, "!@", createTestConfig(endpointUri));
+    public void testUserAgent_usesUnknownAgentWhenProvidedWithBogusAgentString() throws InterruptedException {
+        TestService service = JaxRsClient.create(TestService.class, "bogus version string",
+                createTestConfig(endpointUri));
+        service.string();
+
+        RecordedRequest request = server.takeRequest();
+        assertThat(request.getHeader("User-Agent"), startsWith("unknown/0.0.0"));
+    }
+
+    @Test
+    public void testUserAgent_augmentedByHttpRemotingAndServiceComponents() throws Exception {
+        TestService service = JaxRsClient.create(TestService.class, AGENT, createTestConfig(endpointUri));
+        service.string();
+
+        RecordedRequest request = server.takeRequest();
+        String remotingVersion = OkHttpClients.class.getPackage().getImplementationVersion();
```

Comment:
FYI I'm pretty sure this is always going to be `null` since you're not running tests out of a built jar

Reply:
that's why i thought. but in gradle on circle these are running against jars (that's why the first version of this PR was so red).

Reply:
how is it different in circle than locally?

Reply:
no idea

Reply:
it was green locally, and red on circle. maybe some caching weirdness? i really don't know.